app/src/main/java/io/gnosis/safe/ui/safe/selection/SafeSelectionDialog.kt

```diff
@@ -63,9 +63,13 @@ class SafeSelectionDialog : BaseBottomSheetDialogFragment<DialogSafeSelectionBin
                 is SafeSelectionState.SafeListState -> {
                     adapter.setItems(state.listItems, state.activeSafe)
                     navHandler?.setSafeData(state.activeSafe)
+                    if (state.viewAction == BaseStateViewModel.ViewAction.CloseScreen) {
+                        binding.root.postDelayed({
+                            dismiss()
+                        }, 200)
```

Comment:
What is this delay for?

Reply:
otherwise, you won't see your selection and dialog will be instantly dismissed. This way checkmark is set, you see it, and then the dialog is dismissed